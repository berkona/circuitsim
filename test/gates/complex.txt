{"undoStack":[],"graph":{"0":{"type":"input","name":"A","pos":{"x":0,"y":85.71428571428571},"rect":{"x":0,"y":85.71428571428571,"width":25,"height":20},"pins":[{"adj":[["50",0]]}]},"1":{"type":"input","name":"B","pos":{"x":0,"y":171.42857142857142},"rect":{"x":0,"y":171.42857142857142,"width":25,"height":20},"pins":[{"adj":[["19","0"]]}]},"2":{"type":"input","name":"C","pos":{"x":0,"y":257.1428571428571},"rect":{"x":0,"y":257.1428571428571,"width":25,"height":20},"pins":[{"adj":[["21","0"]]}]},"3":{"type":"input","name":"D","pos":{"x":0,"y":342.85714285714283},"rect":{"x":0,"y":342.85714285714283,"width":25,"height":20},"pins":[{"adj":[["22",0]]}]},"4":{"type":"input","name":"E","pos":{"x":0,"y":428.57142857142856},"rect":{"x":0,"y":428.57142857142856,"width":25,"height":20},"pins":[{"adj":[["23",0]]}]},"5":{"type":"input","name":"F","pos":{"x":0,"y":514.2857142857142},"rect":{"x":0,"y":514.2857142857142,"width":25,"height":20},"pins":[{"adj":[["28","0"]]}]},"6":{"type":"output","name":"Y","pos":{"x":783,"y":300},"rect":{"x":783,"y":300,"width":25,"height":20},"pins":[{"adj":[["37",0]]}]},"7":{"nid":"7","type":"pmos","pos":{"x":150,"y":25},"rect":{"x":150,"y":25,"width":40,"height":43},"pins":[{"adj":[["18","0"]]},{"adj":[["50",0]]},{"adj":[["15","0"]]}]},"8":{"nid":"8","type":"pmos","pos":{"x":275,"y":75},"rect":{"x":275,"y":75,"width":40,"height":43},"pins":[{"adj":[["18","0"]]},{"adj":[["19",0],["48",0]]},{"adj":[["15","0"]]}]},"9":{"nid":"9","type":"pmos","pos":{"x":400,"y":125},"rect":{"x":400,"y":125,"width":40,"height":43},"pins":[{"adj":[["18","0"]]},{"adj":[["20",0]]},{"adj":[["12","0"]]}]},"12":{"nid":"12","type":"pmos","pos":{"x":275,"y":250},"rect":{"x":275,"y":250,"width":40,"height":43},"pins":[{"adj":[["14","0"],["9","2"]]},{"adj":[["25",0]]},{"adj":[["37",0],["39",0]]}]},"13":{"nid":"13","type":"pmos","pos":{"x":150,"y":250},"rect":{"x":150,"y":250,"width":40,"height":43},"pins":[{"adj":[["14",0]]},{"adj":[["24",0],["44","0"]]},{"adj":[["39",0]]}]},"14":{"nid":"14","type":"wire","pos":{"x":239,"y":250},"rect":{"x":231,"y":242,"width":16,"height":16},"pins":[{"adj":[["13","0"],["12","0"],["15","2"]]}]},"15":{"nid":"15","type":"pmos","pos":{"x":200,"y":150},"rect":{"x":200,"y":150,"width":40,"height":43},"pins":[{"adj":[["8","2"],["7","2"]]},{"adj":[["45",0]]},{"adj":[["14","0"]]}]},"18":{"nid":"18","type":"vcc","pos":{"x":275,"y":0},"rect":{"x":275,"y":0,"width":40,"height":29},"pins":[{"adj":[["7","0"],["8","0"],["9","0"]]}]},"19":{"nid":"19","type":"wire","pos":{"x":39,"y":100},"rect":{"x":31,"y":92,"width":16,"height":16},"pins":[{"adj":[["8","1"],["1","0"]]}]},"20":{"nid":"20","type":"wire","pos":{"x":389,"y":225},"rect":{"x":381,"y":217,"width":16,"height":16},"pins":[{"adj":[["9","1"],["21",0],["33","1"]]}]},"21":{"nid":"21","type":"wire","pos":{"x":39,"y":225},"rect":{"x":31,"y":217,"width":16,"height":16},"pins":[{"adj":[["20","0"],["2","0"]]}]},"22":{"nid":"22","type":"wire","pos":{"x":64,"y":350},"rect":{"x":56,"y":342,"width":16,"height":16},"pins":[{"adj":[["3","0"],["45",0]]}]},"23":{"nid":"23","type":"wire","pos":{"x":89,"y":425},"rect":{"x":81,"y":417,"width":16,"height":16},"pins":[{"adj":[["4","0"],["24","0"]]}]},"24":{"nid":"24","type":"wire","pos":{"x":89,"y":275},"rect":{"x":81,"y":267,"width":16,"height":16},"pins":[{"adj":[["13","1"],["23","0"]]}]},"25":{"nid":"25","type":"wire","pos":{"x":264,"y":325},"rect":{"x":256,"y":317,"width":16,"height":16},"pins":[{"adj":[["12","1"],["43",0]]}]},"26":{"nid":"26","type":"wire","pos":{"x":114,"y":325},"rect":{"x":106,"y":317,"width":16,"height":16},"pins":[{"adj":[["28",0],["43",0]]}]},"28":{"nid":"28","type":"wire","pos":{"x":114,"y":525},"rect":{"x":106,"y":517,"width":16,"height":16},"pins":[{"adj":[["26","0"],["5","0"]]}]},"29":{"nid":"29","type":"nmos","pos":{"x":430.25,"y":498},"rect":{"x":430.25,"y":498,"width":40,"height":43},"pins":[{"adj":[["31","2"]]},{"adj":[["48","0"]]},{"adj":[["42",0]]}]},"31":{"nid":"31","type":"nmos","pos":{"x":431.25,"y":441},"rect":{"x":431.25,"y":441,"width":40,"height":43},"pins":[{"adj":[["36",0]]},{"adj":[["49",0]]},{"adj":[["29","0"]]}]},"32":{"nid":"32","type":"nmos","pos":{"x":347.25,"y":438},"rect":{"x":347.25,"y":438,"width":40,"height":43},"pins":[{"adj":[["36",0]]},{"adj":[["47","0"]]},{"adj":[["38",0]]}]},"33":{"nid":"33","type":"nmos","pos":{"x":382.25,"y":366},"rect":{"x":382.25,"y":366,"width":40,"height":43},"pins":[{"adj":[["37",0]]},{"adj":[["20","0"]]},{"adj":[["36",0]]}]},"34":{"nid":"34","type":"nmos","pos":{"x":213.25,"y":360},"rect":{"x":213.25,"y":360,"width":40,"height":43},"pins":[{"adj":[["39",0]]},{"adj":[["43",0]]},{"adj":[["35","0"]]}]},"35":{"nid":"35","type":"nmos","pos":{"x":220.25,"y":429},"rect":{"x":220.25,"y":429,"width":40,"height":43},"pins":[{"adj":[["34","2"]]},{"adj":[["44",0]]},{"adj":[["40","0"]]}]},"36":{"nid":"36","type":"wire","pos":{"x":423,"y":440},"rect":{"x":415,"y":432,"width":16,"height":16},"pins":[{"adj":[["33","2"],["31","0"],["32","0"]]}]},"37":{"nid":"37","type":"wire","pos":{"x":419,"y":292},"rect":{"x":411,"y":284,"width":16,"height":16},"pins":[{"adj":[["33","0"],["6","0"],["12","2"]]}]},"38":{"nid":"38","type":"wire","pos":{"x":386,"y":541},"rect":{"x":378,"y":533,"width":16,"height":16},"pins":[{"adj":[["32","2"],["42",0]]}]},"39":{"nid":"39","type":"wire","pos":{"x":252,"y":292},"rect":{"x":244,"y":284,"width":16,"height":16},"pins":[{"adj":[["34","0"],["12","2"],["13","2"]]}]},"40":{"nid":"40","type":"gnd","pos":{"x":220.75,"y":505},"rect":{"x":220.75,"y":505,"width":40,"height":29},"pins":[{"adj":[["35","2"]]}]},"41":{"nid":"41","type":"gnd","pos":{"x":364.75,"y":566},"rect":{"x":364.75,"y":566,"width":40,"height":29},"pins":[{"adj":[["42",0]]}]},"42":{"nid":"42","type":"wire","pos":{"x":408,"y":541},"rect":{"x":400,"y":533,"width":16,"height":16},"pins":[{"adj":[["41","0"],["29","2"],["38","0"]]}]},"43":{"nid":"43","type":"wire","pos":{"x":214,"y":325},"rect":{"x":206,"y":317,"width":16,"height":16},"pins":[{"adj":[["25","0"],["26","0"],["34","1"]]}]},"44":{"nid":"44","type":"wire","pos":{"x":144,"y":450},"rect":{"x":136,"y":442,"width":16,"height":16},"pins":[{"adj":[["35","1"],["13","1"]]}]},"45":{"nid":"45","type":"wire","pos":{"x":127,"y":172},"rect":{"x":119,"y":164,"width":16,"height":16},"pins":[{"adj":[["15","1"],["22","0"],["46",0]]}]},"46":{"nid":"46","type":"wire","pos":{"x":130,"y":484},"rect":{"x":122,"y":476,"width":16,"height":16},"pins":[{"adj":[["45",0],["47",0]]}]},"47":{"nid":"47","type":"wire","pos":{"x":349,"y":484},"rect":{"x":341,"y":476,"width":16,"height":16},"pins":[{"adj":[["46","0"],["32","1"]]}]},"48":{"nid":"48","type":"wire","pos":{"x":277,"y":523},"rect":{"x":269,"y":515,"width":16,"height":16},"pins":[{"adj":[["8","1"],["29","1"]]}]},"49":{"nid":"49","type":"wire","pos":{"x":65,"y":461},"rect":{"x":57,"y":453,"width":16,"height":16},"pins":[{"adj":[["31","1"],["50",0]]}]},"50":{"nid":"50","type":"wire","pos":{"x":68,"y":47},"rect":{"x":60,"y":39,"width":16,"height":16},"pins":[{"adj":[["49","0"],["0","0"],["7","1"]]}]}},"inputPins":[{"name":"A","nid":"0"},{"name":"B","nid":"1"},{"name":"C","nid":"2"},{"name":"D","nid":"3"},{"name":"E","nid":"4"},{"name":"F","nid":"5"}],"outputPins":[{"name":"Y","nid":"6"}],"nNodes":51,"simType":"transistor","version":"6"}