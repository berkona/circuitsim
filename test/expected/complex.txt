{ "inputs": [ "A", "B", "C", "D", "E", "F" ],
  "outputs": [ "Y" ],
  "rows": 
   [ [ "1", "1", "1", "1", "1", "1", "0" ],
     [ "1", "1", "1", "1", "1", "0", "0" ],
     [ "1", "1", "1", "1", "0", "1", "0" ],
     [ "1", "1", "1", "1", "0", "0", "0" ],
     [ "1", "1", "1", "0", "1", "1", "0" ],
     [ "1", "1", "1", "0", "1", "0", "0" ],
     [ "1", "1", "1", "0", "0", "1", "0" ],
     [ "1", "1", "1", "0", "0", "0", "0" ],
     [ "1", "1", "0", "1", "1", "1", "0" ],
     [ "1", "1", "0", "1", "1", "0", "1" ],
     [ "1", "1", "0", "1", "0", "1", "1" ],
     [ "1", "1", "0", "1", "0", "0", "1" ],
     [ "1", "1", "0", "0", "1", "1", "0" ],
     [ "1", "1", "0", "0", "1", "0", "1" ],
     [ "1", "1", "0", "0", "0", "1", "1" ],
     [ "1", "1", "0", "0", "0", "0", "1" ],
     [ "1", "0", "1", "1", "1", "1", "0" ],
     [ "1", "0", "1", "1", "1", "0", "0" ],
     [ "1", "0", "1", "1", "0", "1", "0" ],
     [ "1", "0", "1", "1", "0", "0", "0" ],
     [ "1", "0", "1", "0", "1", "1", "0" ],
     [ "1", "0", "1", "0", "1", "0", "1" ],
     [ "1", "0", "1", "0", "0", "1", "1" ],
     [ "1", "0", "1", "0", "0", "0", "1" ],
     [ "1", "0", "0", "1", "1", "1", "0" ],
     [ "1", "0", "0", "1", "1", "0", "1" ],
     [ "1", "0", "0", "1", "0", "1", "1" ],
     [ "1", "0", "0", "1", "0", "0", "1" ],
     [ "1", "0", "0", "0", "1", "1", "0" ],
     [ "1", "0", "0", "0", "1", "0", "1" ],
     [ "1", "0", "0", "0", "0", "1", "1" ],
     [ "1", "0", "0", "0", "0", "0", "1" ],
     [ "0", "1", "1", "1", "1", "1", "0" ],
     [ "0", "1", "1", "1", "1", "0", "0" ],
     [ "0", "1", "1", "1", "0", "1", "0" ],
     [ "0", "1", "1", "1", "0", "0", "0" ],
     [ "0", "1", "1", "0", "1", "1", "0" ],
     [ "0", "1", "1", "0", "1", "0", "1" ],
     [ "0", "1", "1", "0", "0", "1", "1" ],
     [ "0", "1", "1", "0", "0", "0", "1" ],
     [ "0", "1", "0", "1", "1", "1", "0" ],
     [ "0", "1", "0", "1", "1", "0", "1" ],
     [ "0", "1", "0", "1", "0", "1", "1" ],
     [ "0", "1", "0", "1", "0", "0", "1" ],
     [ "0", "1", "0", "0", "1", "1", "0" ],
     [ "0", "1", "0", "0", "1", "0", "1" ],
     [ "0", "1", "0", "0", "0", "1", "1" ],
     [ "0", "1", "0", "0", "0", "0", "1" ],
     [ "0", "0", "1", "1", "1", "1", "0" ],
     [ "0", "0", "1", "1", "1", "0", "0" ],
     [ "0", "0", "1", "1", "0", "1", "0" ],
     [ "0", "0", "1", "1", "0", "0", "0" ],
     [ "0", "0", "1", "0", "1", "1", "0" ],
     [ "0", "0", "1", "0", "1", "0", "1" ],
     [ "0", "0", "1", "0", "0", "1", "1" ],
     [ "0", "0", "1", "0", "0", "0", "1" ],
     [ "0", "0", "0", "1", "1", "1", "0" ],
     [ "0", "0", "0", "1", "1", "0", "1" ],
     [ "0", "0", "0", "1", "0", "1", "1" ],
     [ "0", "0", "0", "1", "0", "0", "1" ],
     [ "0", "0", "0", "0", "1", "1", "0" ],
     [ "0", "0", "0", "0", "1", "0", "1" ],
     [ "0", "0", "0", "0", "0", "1", "1" ],
     [ "0", "0", "0", "0", "0", "0", "1" ] ] }